File: problems/2506-count-pairs-of-similar-strings/solution.py
class Solution:
    def similarPairs(self, words: List[str]) -> int:
        count = 0
        for i in range(len(words)):
            for j in range(i + 1, len(words)):
                set1 = set(words[i])
                set2 = set(words[j])
                if set1 == set2:
                    count += 1
        return count

File: problems/2506-count-pairs-of-similar-strings/solution.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int similarPairs(String[] words) {
        int count = 0;
        for (int i = 0; i < words.length; i++) {
            for (int j = i + 1; j < words.length; j++) {
                if (areSimilar(words[i], words[j])) {
                    count++;
                }
            }
        }
        return count;
    }

    private boolean areSimilar(String word1, String word2) {
        Set<Character> set1 = new HashSet<>();
        Set<Character> set2 = new HashSet<>();

        for (char c : word1.toCharArray()) {
            set1.add(c);
        }
        for (char c : word2.toCharArray()) {
            set2.add(c);
        }

        return set1.equals(set2);
    }
}